lib/src/install/config.rs

```diff
@@ -171,10 +197,20 @@ pub(crate) fn load_config() -> Result<Option<InstallConfiguration>> {
         if let Some(config) = config.as_mut() {
             if let Some(install) = c.install {
                 tracing::debug!("Merging install config: {install:?}");
-                config.merge(install);
+                config.merge(install, &env);
             }
         } else {
-            config = c.install;
+            // Only set the config if it matches the current arch
+            // If no arch is specified, set the config unconditionally
+            if let Some(ref mut install) = c.install {
+                if let Some(arch) = install.match_architectures.as_ref() {
+                    if arch.contains(&env.sys_arch) {
```

Comment:
I think this can boil down to:

```
            if let Some(ref mut install) = c.install {
                if install.match_architectures.as_ref().map(|a| a.contains(&env.sys_arch)).unwrap_or(true) {
```
                    if  {